quil-rs/src/reserved.rs

```diff
@@ -13,6 +13,7 @@ pub enum ReservedToken {
 }
 
 #[derive(Clone, Debug)]
+#[allow(dead_code)] // clippy 0.1.77 incorrectly identifies this NewType as dead code.
 pub struct NotReservedToken(String);
```

Comment:
More specifically, it's that the inner (private) `String` is never accessed.

Since this is used as an error, it should implement `Error`:

```suggestion
#[derive(Clone, Debug, thiserror::Error)]
#[error("{0} is not a reserved token")]
pub struct NotReservedToken(String);
```

Reply:
If the above suggestion is applied, also need to change the `Err` case below to `Err(NotReservedToken(s.to_string()))`